### ISSUE TITLE
Propagate task exceptions?

### ISSUE THREAD

**Issue by wgordon17** [coleifer/huey#638]
Currently, I have a mechanism where 1 task (TaskA) calls another task (TaskB). I have an `on_error` handler for TaskA, but when the exception occurs in TaskB, it doesn't get handled by the handler for TaskA.

Is this by design, or am I configuring things incorrectly?

FWIW, my use case is for long running command line scripts. I have a prerequisite where I have to download the necessary CLI tools, and then run those tools. So I have my task set up to

1. Call TaskA
2. TaskA calls TaskB to download tool 1
3. TaskA calls TaskB again to download tool 2 (that way tool 1 and tool 2 can download concurrently)
4. TaskA then blocks waiting on both TaskB results
5. TaskA then continues by running the necessary CLI commands with tool 1 and tool 2

I understand I could _probably_ just rearchitect my pipelines so that I call TaskB outside of TaskA, _then_ call TaskA. I've also noticed that if I set up my pipelines with a `.error()`, the `.error()` only applies to the _last_ task, as opposed to being an error handler for _all_ tasks in the pipeline.

**Comment by coleifer** [comment]
The error handlers are just tasks that run when the task they're bound-to encounters an exception. It's a very simple mechanism.

I'd probably write the task like this:

```python
@huey.task()
def task_a():
    b1 = task_b()
    b2 = task_b()
    try:
        r1, r2 = b1.get(True), b2.get(True)
    except Exception as exc:
        # Do your error handling here.
        ...
    ... do something now that b1/b2 are done ...
```
